tensorboard/summary/writer/event_file_writer.py

```diff
@@ -165,20 +165,33 @@ def __init__(self, record_writer, max_queue_size=20, flush_secs=120):
     def write(self, bytestring):
         """Enqueue the given bytes to be written asychronously."""
         with self._lock:
+            # Status of the worker should be checked under the lock to avoid
+            # multiple threads passing the check and then switching just before
+            # blocking on putting to the queue which might result in a deadlock.
+            self._check_worker_status()
             if self._closed:
                 raise IOError("Writer is closed")
             self._byte_queue.put(bytestring)
+            # Check the status again in case the background worker thread has
+            # failed in the meantime to avoid waiting until the next call to
+            # surface the error.
+            self._check_worker_status()
 
     def flush(self):
         """Write all the enqueued bytestring before this flush call to disk.
 
         Block until all the above bytestring are written.
         """
+        self._check_worker_status()
```

Comment:
Could this be moved under the lock?

Reply:
Good point, it might be relevant to the problem of flushing, thanks! Done